src/test/unit/engine/CppRegionTest.cpp

```diff
@@ -168,7 +168,6 @@ TEST(CppRegionTest, testYAML) {
 
   Network net;
   std::shared_ptr<Region> level1;
-  EXPECT_THROW(net.addRegion("level1", "TestNode", badparams), exception);
```

Comment:
why is this removed?

Reply:
> why is this removed?      (talking about the check for bad parameters)

The new behavior is that we do not use the Spec to evaluate the parameters so we have no way to know if 'unexpected' parameters are presented.

It is possible to restore the original behavior....on a per Region basis.  In the base class for all Region implementations `RegionImpl.hpp`  there is a new function that can be called to pre-process the parameters.
```
  virtual ValueMap ValidateParameters(const ValueMap &vm, Spec* ns);
```
This will read the Spec, insert the default values into the ValueMap, and validate for unexpected parameters. Each implementation has the option to call this in its constructor before it handles the parameters.  In an attempt to simplify everything I am first trying to do parameter processing without using this function but it is there as a fallback.

Reply:
> In an attempt to simplify everything I am first trying to do parameter processing without using this function but it is there as a fallback.
 
i think it's fine not to check that, so do as you do now. 

> so we have no way to know if 'unexpected' parameters are presented.

just what happens if I pass a param `canIBreakStuff = True` to a SPRegion?

Reply:
> just what happens if I pass a param canIBreakStuff = True to a SPRegion?

It will be ignored.